arwen/host/arwen.go

```diff
@@ -52,6 +52,7 @@ type vmHost struct {
 // NewArwenVM creates a new Arwen vmHost
 func NewArwenVM(
 	blockChainHook vmcommon.BlockchainHook,
+	addressGenerator arwen.AddressGenerator,
```

Comment:
The address generator must be passed as part of `VMHostParameters`. Please do not alter the constructor signature.